### PR TITLE
update: separate page for the common ViewModel

### PR DESCRIPTION
**Pull request by zamulla**

### CODE REVIEW

File: topics/compose/compose-viewmodel.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    @Composable
+    fun CupcakeApp(
+        viewModel: OrderViewModel = viewModel { OrderViewModel() },
+    ) {
+    // ...
+    }
+    ```
+   
+On non-JDK platforms objects cannot be instantiated using type reflection.
+So in common code you cannot call the `viewModel()` function without parameters: every time a ViewModel is created,
```
Comment (MatkovIvan) [inline topics/compose/compose-viewmodel.md:58]:
```suggestion
So in common code you cannot call the `viewModel()` function without parameters: every time a `ViewModel` is created,
```

File: topics/compose/compose-viewmodel.md [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+you need to provide at least an initializer as an argument.
+
+If only an initializer is provided, Compose Multiplatform creates a default factory under the hood.
```
Comment (MatkovIvan) [inline topics/compose/compose-viewmodel.md:61]: > Compose Multiplatform creates

it's not correct. It does not relate to Compose

**Review by MatkovIvan — CHANGES_REQUESTED** [review]

File: topics/compose/compose-viewmodel.md [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    ```
+
+2. Add the ViewModel to your composable function:
+
+    ```kotlin
+    @Composable
+    fun CupcakeApp(
+        viewModel: OrderViewModel = viewModel { OrderViewModel() },
+    ) {
+    // ...
```
Comment (MatkovIvan) [inline topics/compose/compose-viewmodel.md:53]:
```suggestion
        // ...
```

File: topics/compose/compose-viewmodel.md [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    }
+    ```
+   
+On non-JDK platforms objects cannot be instantiated using type reflection.
+So in common code you cannot call the `viewModel()` function without parameters: every time a `ViewModel` is created,
+you need to provide at least an initializer as an argument.
+
+If only an initializer is provided, the library creates a default factory under the hood.
+But you can implement your own factories and call more explicit versions of the common `viewModel()` function,
```
Comment (MatkovIvan) [inline topics/compose/compose-viewmodel.md:62]:
```suggestion
But you can implement your own factories and call more explicit versions of the common `viewModel` function,
```

File: topics/compose/compose-viewmodel.md [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+To use the multiplatform `ViewModel` implementation, add the following dependency to your `commonMain` source set:
+
+```kotlin
+kotlin {
+  // ...
+  sourceSets {
+    // ...
+    commonMain.dependencies {
+      // ...
+      implementation("org.jetbrains.androidx.lifecycle:lifecycle-viewmodel-compose:%composeViewmodelVersion%")
```
Comment (MatkovIvan) [inline topics/compose/compose-viewmodel.md:21]: `%composeViewmodelVersion%` should be updated to `2.8.0`

**Review by MatkovIvan — APPROVED** [review]

**Review by koshachy — APPROVED** [review]
Approved.
Resolve the comments before merging.

**Pull request merged**